matminer/featurizers/site.py

```diff
@@ -413,24 +413,32 @@ def from_preset(preset, **kwargs):
                 else:
                     op_types[k + 1] = ["wt"]
 
-            return CrystalNNFingerprint(op_types, **kwargs)
+            return CrystalNNFingerprint(op_types, chem_info=None, **kwargs)
 
         else:
             raise RuntimeError('preset "{}" is not supported in '
                                'CrystalNNFingerprint'.format(preset))
 
-    def __init__(self, op_types, **kwargs):
+    def __init__(self, op_types, chem_info=None, **kwargs):
         """
         Initialize the CrystalSiteFingerprint. Use the from_preset() function to
         use default params.
         Args:
             op_types (dict): a dict of coordination number (int) to a list of str
                 representing the order parameter types
+            chem_info (dict): a dict of chemical properties (e.g., atomic mass)
```

Comment:
In practice, I think people are going to have a hard time passing in the chem_info dict and find this to be a pain to use. While flexible, normally you have a dataset of 1000 compounds and don't want to manually construct a dictionary of all possible species to some value like electronegativity.

It might be better if this was able to easily work with an "AbstractData" class instead, since those already have a bunch of built-in dictionaries for various properties. Or, some easy way for the user to leverage those classes rather than construct the dictionary themselves.

Or, maybe instead AbstractData itself could implement a method that turns its properties into a dictionary. That way you could do ``PymatgenData.dict_output("X")`` that will turn all the PymatgenData electronegativities to a dictionary format.